woocommerce/woocommerce-ios
# [Woo POS][Cash & Receipts] Complete order paid by cash

Closes: #14602

## Description
This PR adds the logic to mark orders as completed when paid by cash via the POS. Only tackles part of the happy path for the moment. We still have not implemented:
- Error handling/change due
- Redirect to success view upon completion

## Changes: 
- Updates `PointOfSaleOrderController` with a new `collectCashPayment()` method
- Dispatches `OrderAction.updateOrder` when the in-flight order is marked as completed.
- Updates existing `FormattableAmountTextField` with new `.pos` style, so we can reuse it.

## Testing
1. Enable `.acceptCashForPointOfSale`
2. Process a POS transaction via tapping the `Cash Payment" button
3. Observe the built-in keyboard can be used
4. Enter any amount in the text field. This reuses the `FormattableAmountTextField` component which already takes into account locale an currency.
5. Tap `Mark payment as completed`. Observe the loading indicator and the button being disabled while loading.
6. Once done, close the POS, go back to the app's Orders list, and observe the new order is there, marked complete and with `Cash (Point of Sale)` message (this is TBD right now).

https://github.com/user-attachments/assets/4528a047-8eac-45e9-b7ae-980195e81a6f

## Feedback:
I have not implemented these bits yet, so any advice is appreciated:
- Does it feel adequate to have the collect payment logic within the Order Controller? I think we can add here the error handling as well as the different methods needed for validating the cash amounts.
- Currently the presentation of the success view relies entirely on `CardPresentPaymentEventDetails.paymentSuccess`, but this doesn't make much sense if we accept cash. I'm thinking we should intercept `PointOfSaleAggregateModel.publishPaymentMessages` with a new publisher for cash, so we can not only present the success view but also cancel payment intents when cash is selected. Thoughts?

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
